### DISCUSSION TITLE
Trying to install coc-metals

### DISCUSSION THREAD

**Discussion by BeLeap** [scalameta/coc-metals#391] [Q&A]
Hello, I'm trying to install coc-metals. If you turn on the scala file after installing coc-metals, coc-metals appear to be attempting to download metals. However, while downloading metals, the message Failed to download Metals appears. Please refer to the attached image of Java installation location of my computer and message below.
![image](https://user-images.githubusercontent.com/46488521/113325895-f3654100-9353-11eb-8670-30e2d34565d7.png)
![image](https://user-images.githubusercontent.com/46488521/113325923-fb24e580-9353-11eb-81e2-9922627fe141.png)

=== Environments ===
OS: Windows 10 20H2
NeoVim: v0.5.0
coc.nvim: maybe latest...idk(Updated Plugin before installing coc-metals)
coc-metals: v1.0.2

**Comment by ckipp01** [comment]
Thanks for the report @BeLeap. Are you by chanced behind a proxy or firewall or anything? Also, that location for your Java home, are you passing it into `coc-metals` via the `metals.javaHome` setting, or is that just the one that it's picking up?

**Reply by BeLeap** [reply]
I'm not sure, but I don't use a proxy. I tried both the environment variable and the one in CocConfig, but neither of them worked.

**Reply by ckipp01** [reply]
🤔 if you aren't behind a proxy or anything then this should work as expected. By "environment variable" do you mean having `JAVA_HOME` set?

**Reply by BeLeap** [reply]
Yes
![image](https://user-images.githubusercontent.com/46488521/113336220-06324280-9361-11eb-8506-3cc566de2502.png)

**Reply by ckipp01** [reply]
Hmm, I'm not fully sure what's going on. Behind the scenes here coursier is attempting to bootstrap Metals. Do you by chance have [coursier](https://get-coursier.io/) installed? If so, we could try to test and see if doing

```
coursier bootstrap \
  --java-opt -Xss4m \
  --java-opt -Xms100m \
  org.scalameta:metals_2.12:0.10.0 \
  -r bintray:scalacenter/releases \
  -r sonatype:snapshots \
  -o /usr/local/bin/metals -f
```
Just to see if can actually download metals. This will at least tell us if it's an issue downloading or an issue with your Java version.

**Reply by BeLeap** [reply]
![image](https://user-images.githubusercontent.com/46488521/113479203-a5257e80-94c8-11eb-8e4e-491114a3341b.png)
It throws error

**Reply by ckipp01** [reply]
🤔 huh, if that is throwing an error, then for sure the actual download inside of the extension won't work. I don't believe I've actually seen this error before. @tgodzik if you have a sec, does this look familiar to you at all?

**Reply by tgodzik** [reply]
`/usr/local/bin/metals ` is a linux path, so most likely you need an alternative one for windows.

**Reply by ckipp01** [reply]
Ach duh. @BeLeap are you able to adjust the `-o` path and give it another shot?

**Reply by BeLeap** [reply]
![image](https://user-images.githubusercontent.com/46488521/113480145-ea987a80-94cd-11eb-864f-5a23a2064928.png)
It runs well

**Reply by ckipp01** [reply]
ah ok, so then this does point to something going wrong with coc-metals trying to correctly get your Java home. Seeing the very first image, what you have set has your `JAVA_HOME` should work. And you said even when you explicitly try setting that in your `:CocConfig` it also fails?

**Reply by BeLeap** [reply]
I've set the JDK location on metals.javaHome, but it didn't work.

**Reply by BeLeap** [reply]
![image](https://user-images.githubusercontent.com/46488521/113480382-f0429000-94ce-11eb-8832-8f31c39d56b9.png)
![image](https://user-images.githubusercontent.com/46488521/113480391-f59fda80-94ce-11eb-80d5-fdcb52737a05.png)
The results I've just tried are shown in the above.
